2017-09-06T15:35:02,130 [ERROR] (    AbstractJDBCDao.java: 290) - Cannot parse Object 
java.sql.SQLException: no such column: 'Course_id'
	at org.sqlite.RS.findColumn(RS.java:121) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.RS.getInt(RS.java:293) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.parseDependenceResultSet(AbstractJDBCDao.java:287) [bin/:?]
	at main.dao.AbstractJDBCDao.getInstanceallDepencedObj1(AbstractJDBCDao.java:141) [bin/:?]
	at test.dao.CourseDAOTest.testgetInstanceallDepencedObj1(CourseDAOTest.java:77) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T15:37:20,904 [ERROR] (    AbstractJDBCDao.java: 290) - Cannot parse Object 
java.sql.SQLException: no such column: 'Course_id'
	at org.sqlite.RS.findColumn(RS.java:121) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.RS.getInt(RS.java:293) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.parseDependenceResultSet(AbstractJDBCDao.java:287) [bin/:?]
	at main.dao.AbstractJDBCDao.getInstanceallDepencedObj1(AbstractJDBCDao.java:141) [bin/:?]
	at test.dao.CourseDAOTest.testgetInstanceallDepencedObj1(CourseDAOTest.java:77) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T17:38:39,556 [ERROR] (    AbstractJDBCDao.java: 174) - Cannot return any Object 
java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such table: Group)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.throwex(DB.java:374) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.NativeDB.prepare(Native Method) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.prepare(DB.java:123) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.<init>(PrepStmt.java:42) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:404) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:399) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.getAll(AbstractJDBCDao.java:165) [bin/:?]
	at test.dao.GroupDAOTest.testGetAll(GroupDAOTest.java:43) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T17:42:59,308 [ERROR] (    AbstractJDBCDao.java: 174) - Cannot return any Object 
java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such table: Group)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.throwex(DB.java:374) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.NativeDB.prepare(Native Method) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.prepare(DB.java:123) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.<init>(PrepStmt.java:42) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:404) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:399) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.getAll(AbstractJDBCDao.java:165) [bin/:?]
	at test.dao.GroupDAOTest.testGetAll(GroupDAOTest.java:43) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:47:02,004 [ERROR] (    AbstractJDBCDao.java: 202) - Cannot insert Object 
java.sql.SQLException: [SQLITE_BUSY]  The database file is locked (database is locked)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:339) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.persist(AbstractJDBCDao.java:173) [bin/:?]
	at main.dao.SqLiteCourseDao.create(SqLiteCourseDao.java:46) [bin/:?]
	at test.dao.CourseDAOTest.testGetByPK(CourseDAOTest.java:57) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:47:05,029 [ERROR] (    AbstractJDBCDao.java: 202) - Cannot insert Object 
java.sql.SQLException: [SQLITE_BUSY]  The database file is locked (database is locked)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:339) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.persist(AbstractJDBCDao.java:173) [bin/:?]
	at main.dao.SqLiteCourseDao.create(SqLiteCourseDao.java:46) [bin/:?]
	at test.dao.CourseDAOTest.testCreate(CourseDAOTest.java:45) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:47:08,036 [ERROR] (    AbstractJDBCDao.java: 202) - Cannot insert Object 
java.sql.SQLException: [SQLITE_BUSY]  The database file is locked (database is locked)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:339) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.persist(AbstractJDBCDao.java:173) [bin/:?]
	at main.dao.SqLiteCourseDao.create(SqLiteCourseDao.java:46) [bin/:?]
	at test.dao.CourseDAOTest.testDelete(CourseDAOTest.java:113) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:47:08,047 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such table: ProfessorProfessors_Courses)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.throwex(DB.java:374) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.NativeDB.prepare(Native Method) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.prepare(DB.java:123) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.<init>(PrepStmt.java:42) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:404) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:399) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceAllDependentObj(AbstractJDBCDao.java:269) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceAllDependentObj(CourseDAOTest.java:91) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:47:11,054 [ERROR] (    AbstractJDBCDao.java: 257) - Cannot update Object Course4 [name=Math, duration=12, groups=0, professors=0]
java.sql.SQLException: [SQLITE_BUSY]  The database file is locked (database is locked)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:339) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.update(AbstractJDBCDao.java:247) [bin/:?]
	at test.dao.CourseDAOTest.testUpdate(CourseDAOTest.java:138) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:51:18,392 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such table: ProfessorProfessors_Courses)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.throwex(DB.java:374) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.NativeDB.prepare(Native Method) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.prepare(DB.java:123) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.<init>(PrepStmt.java:42) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:404) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:399) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.Conn.prepareStatement(Conn.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceAllDependentObj(AbstractJDBCDao.java:269) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceAllDependentObj(CourseDAOTest.java:91) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T20:57:39,212 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_CONSTRAINT]  Abort due to constraint violation (columns professor_id, course_id are not unique)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:342) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceDependentObj(AbstractJDBCDao.java:273) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceDependentProfessor(CourseDAOTest.java:91) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T21:13:52,534 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_CONSTRAINT]  Abort due to constraint violation (columns group_id, course_id are not unique)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:342) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceDependentObj(AbstractJDBCDao.java:273) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceDependentGroup(CourseDAOTest.java:106) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T21:13:52,718 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_CONSTRAINT]  Abort due to constraint violation (columns professor_id, course_id are not unique)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:342) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceDependentObj(AbstractJDBCDao.java:273) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceDependentProfessor(CourseDAOTest.java:91) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-06T21:16:05,713 [ERROR] (    AbstractJDBCDao.java: 284) - Cannot insert Object 
java.sql.SQLException: [SQLITE_CONSTRAINT]  Abort due to constraint violation (columns group_id, course_id are not unique)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:342) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.addInstanceDependentObj(AbstractJDBCDao.java:273) [bin/:?]
	at test.dao.CourseDAOTest.testAddInstanceDependentGroup(CourseDAOTest.java:106) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T16:31:24,889 [ERROR] (    AbstractJDBCDao.java: 176) - Cannot insert Object 
java.sql.SQLException: [SQLITE_BUSY]  The database file is locked (database is locked)
	at org.sqlite.DB.newSQLException(DB.java:383) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.newSQLException(DB.java:387) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.execute(DB.java:339) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.DB.executeUpdate(DB.java:363) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.executeUpdate(PrepStmt.java:85) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.persist(AbstractJDBCDao.java:147) [bin/:?]
	at main.dao.SqLiteCourseDao.create(SqLiteCourseDao.java:48) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:159) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T16:31:53,559 [ERROR] (    AbstractJDBCDao.java: 206) - Cannot delete Object Course109 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T16:38:05,911 [ERROR] (    AbstractJDBCDao.java: 206) - Cannot delete Object Course110 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T16:47:37,675 [ERROR] (    AbstractJDBCDao.java: 206) - Cannot delete Object Course112 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T17:01:55,822 [ERROR] (    AbstractJDBCDao.java: 208) - Cannot delete Object Course121 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-08T17:01:55,902 [ERROR] (    AbstractJDBCDao.java: 208) - Cannot delete Object Course124 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDelete(CourseDAOTest.java:143) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-12T19:06:36,091 [ERROR] (    AbstractJDBCDao.java: 208) - Cannot delete Object Course133 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-12T19:06:36,520 [ERROR] (    AbstractJDBCDao.java: 208) - Cannot delete Object Course136 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:192) [bin/:?]
	at test.dao.CourseDAOTest.testDelete(CourseDAOTest.java:143) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-12T19:11:13,538 [ERROR] (    AbstractJDBCDao.java: 210) - Cannot delete Object Course137 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:194) [bin/:?]
	at test.dao.CourseDAOTest.testDeletefromCourseGroup(CourseDAOTest.java:182) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
2017-09-12T19:11:13,956 [ERROR] (    AbstractJDBCDao.java: 210) - Cannot delete Object Course140 [name=JavaEE, duration=10, groups=0, professors=0]
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.sqlite.PrepStmt.batch(PrepStmt.java:173) ~[sqlite-jdbc-3.7.2.jar:?]
	at org.sqlite.PrepStmt.setObject(PrepStmt.java:226) ~[sqlite-jdbc-3.7.2.jar:?]
	at main.dao.AbstractJDBCDao.delete(AbstractJDBCDao.java:194) [bin/:?]
	at test.dao.CourseDAOTest.testDelete(CourseDAOTest.java:143) [bin/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_144]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit.jar:?]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit.jar:?]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit.jar:?]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit.jar:?]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit.jar:?]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit.jar:?]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit.jar:?]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit.jar:?]
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit.jar:?]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit.jar:?]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit.jar:?]
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit.jar:?]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit.jar:?]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit.jar:?]
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382) [.cp/:?]
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192) [.cp/:?]
